rllib/env/single_agent_env_runner.py

```diff
@@ -320,7 +327,8 @@ def _sample(
             actions = to_env.pop(Columns.ACTIONS)
             actions_for_env = to_env.pop(Columns.ACTIONS_FOR_ENV, actions)
             # Try stepping the environment.
-            results = self._try_env_step(actions_for_env)
+            with self.metrics.log_time(ENV_STEP_TIMER):
```

Comment:
The metrics context can go inside `_try_env_step` so we don't have to duplicate the call to it for single- and multi-agent env runners.

Reply:
That's a good idea. Let's move it to the parent.

Reply:
Ah, wait. I just see that the base class does not know a `MetricsLogger`. I would like to keep the `EnvRunner` itself free from the `MetricsLogger`. We might instead pass the metrics logger in as an argument - still not very clean and increases dependence.